### PR TITLE
[LPF-551]: Create complex unique id's for reports

### PR DESCRIPTION
**Pull request by RobertBuczek**
This pull request replaces the current use of `Integer` as IDS with `UUIDs` for all entities within the reports and mapping tables.

Key Changes:
- Migration: Existing entities with Integer IDs migrated to use UUIDs as their primary keys.
- Code Refactoring: All database interactions (e.g., queries, updates, insertions) are adjusted to accommodate the change from Integer to UUID.
- Testing: Unit and integration tests have been updated

### CODE REVIEW

File: src/test/java/uk/gov/laa/gpfd/dao/ReportTrackingTableDAOTest.java [hunk 1]
```diff
@@ -14,13 +14,16 @@
 import java.time.LocalDateTime;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/dao/ReportTrackingTableDAOTest.java:19]: remove wildcard as this will get flagged by sonarcloud

File: src/test/java/uk/gov/laa/gpfd/integration/ServerSideErrorIT.java [hunk 2]
```diff
@@ -41,24 +41,24 @@ class ServerSideErrorIT {
     AzureGraphClient mockAzureGraphClient;
 
     @Test
-    void getReportsShouldReturn500WhenCannotConnectToDb() throws Exception {
+    void getReportsShouldReturn404WhenCannotConnectToDb() throws Exception {
         MockHttpServletResponse response =  mockMvc.perform(get("/reports")
                 .contentType(MediaType.APPLICATION_JSON)).andReturn().getResponse();
 
-        Assertions.assertEquals(500, response.getStatus());
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/integration/ServerSideErrorIT.java:48]: What's the reason these no longer return 500?

File: src/test/java/uk/gov/laa/gpfd/model/ReportTrackingTableTest.java [hunk 3]
```diff
@@ -158,33 +161,33 @@ void testReportTrackingTableIdBoundaryValues() {
...
         // When
         // Then
-        assertEquals(Integer.MIN_VALUE, table1.id());
-        assertEquals(Integer.MAX_VALUE, table2.id());
+        assertEquals(id, table1.id());
+        assertEquals(id, table2.id());
     }
 
     @Test
     void testReportTrackingTableMappingIdBoundaryValues() {
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/model/ReportTrackingTableTest.java:177]: This test is no longer needed as there is no min and max now we're using UUIDS

File: src/test/java/uk/gov/laa/gpfd/model/ReportTrackingTableTest.java [hunk 4]
```diff
@@ -158,33 +161,33 @@ void testReportTrackingTableIdBoundaryValues() {
         // Given
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/model/ReportTrackingTableTest.java:161]: Test is no longer needed as there are not boundary values now we're using UUIDs

File: src/test/java/uk/gov/laa/gpfd/service/MappingTableServiceTest.java [hunk 5]
```diff
@@ -13,10 +13,7 @@
 import uk.gov.laa.gpfd.model.MappingTable;
 import uk.gov.laa.gpfd.services.MappingTableService;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/service/MappingTableServiceTest.java:16]: Remove wildcard

File: src/main/java/uk/gov/laa/gpfd/services/MappingTableService.java [hunk 6]
```diff
@@ -60,15 +61,8 @@ public List<ReportsGet200ResponseReportListInner> fetchReportListEntries() {
...
-        MappingTable reportListResponse = null;
-
-        if (requestedId < 1000 && requestedId > 0) {
-            reportListResponse = mappingTableDao.fetchReport(requestedId);
-
-        } else {
-            throw new IndexOutOfBoundsException("Report ID needs to be a number between 0 and 1000");
-        }
+    public MappingTable getDetailsForSpecificMapping(UUID requestedId)  {
+        MappingTable reportListResponse = mappingTableDao.fetchReport(requestedId);
```
Comment (ChrisCookOC) [inline src/main/java/uk/gov/laa/gpfd/services/MappingTableService.java:65]: Function documentation above is out of date now

File: src/main/java/uk/gov/laa/gpfd/services/ReportService.java [hunk 7]
```diff
@@ -75,7 +76,7 @@ public ByteArrayOutputStream createCsvStream(String sqlQuery) throws IOException
      * @param requestedId - the ID of the requested report
      * @return a ResponseEntity of type 'StreamingResponseBody', containing a stream of CSV data
      */
-    public ResponseEntity<StreamingResponseBody> createCSVResponse(int requestedId) throws ReportIdNotFoundException, DatabaseReadException, IndexOutOfBoundsException, CsvStreamException {
+    public ResponseEntity<StreamingResponseBody> createCSVResponse(UUID requestedId) throws ReportIdNotFoundException, DatabaseReadException, IndexOutOfBoundsException, CsvStreamException {
```
Comment (ChrisCookOC) [inline src/main/java/uk/gov/laa/gpfd/services/ReportService.java:79]: This can no longer throw an IndexOutOfBoundsException

File: src/main/java/uk/gov/laa/gpfd/services/ReportService.java [hunk 8]
```diff
@@ -113,7 +114,7 @@ public ResponseEntity<StreamingResponseBody> createCSVResponse(int requestedId)
      * @throws ReportIdNotFoundException - From the getDetailsForSpecificReport() method call, if the requested index is not found
      * @throws DatabaseReadException     - From the createReportListResponseList() method call inside getDetailsForSpecificReport()
      */
-    public GetReportById200Response createReportResponse(int id) throws IndexOutOfBoundsException {
+    public GetReportById200Response createReportResponse(UUID id) throws IndexOutOfBoundsException {
```
Comment (ChrisCookOC) [inline src/main/java/uk/gov/laa/gpfd/services/ReportService.java:117]: This can no longer throw an IndexOutOfBoundsException

File: src/test/java/uk/gov/laa/gpfd/dao/ReportTrackingTableDAOTest.java (hunk 1 above)
Comment (ChrisCookOC) [reply src/test/java/uk/gov/laa/gpfd/dao/ReportTrackingTableDAOTest.java:19]: It doesn't currently - may need to investigate the setup if we want it to :) 

https://sonarcloud.io/project/issues?id=ministryofjustice_payforlegalaid&pullRequest=107&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true

File: src/test/java/uk/gov/laa/gpfd/controller/ReportsControllerTest.java [hunk 9]
```diff
@@ -43,19 +44,21 @@
 @ActiveProfiles("testauth")
 class ReportsControllerTest {
 
-    @MockitoBean
+    public static final UUID DEFAULT_ID = UUID.fromString("0d4da9ec-b0b3-4371-af10-f375330d85d1");
+
+    @MockBean
```
Comment (ChrisCookOC) [inline src/test/java/uk/gov/laa/gpfd/controller/ReportsControllerTest.java:49]: MockBean will be deprecated when you merge in the pom changes that pushes SpringBoot to 3.4

**Review by ChrisCookOC — APPROVED** [review]
